test/jobs/test_batch_autoscaling.py

```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/env python
```

Comment:
This looks like a lot of carefully written test code.   However, I don't understand how it's supposed to improve my confidence in the correctness of the new functionality, considering it's harder (for me) to read or understand than the functionality it's testing.  What kind of bugs exactly is it supposed to catch?

Reply:
Also - what kinds of non-bug changes is it supposed to tolerate?